### PR TITLE
Add viability checks for various moves

### PR DESCRIPTION
**Pull request by KrisXV**

### CODE REVIEW

File: js/search.js [hunk 1]
```diff
@@ -1093,10 +1093,12 @@
 				var isViable = BattleMovedex[id] && BattleMovedex[id].isViable;
 				if (id === 'aerialace') isViable = (toID(set.species) in {scyther:1, aerodactylmega:1, kricketune:1});
 				if (id === 'ancientpower') isViable = (toID(set.ability) === 'technician' || (toID(set.ability) === 'serenegrace') || (template.types.indexOf('rock') > 0 && moves.indexOf('powergem') < 0));
+				if (id === 'aurawheel') isViable = (toID(set.species).startsWith('morpeko'));
```
Comment (Zarel) [inline js/search.js]: Isn't it always viable? Just add the flag to moves.js.

File: js/search.js [hunk 2]
```diff
@@ -1121,10 +1124,11 @@
 				if (id === 'skyattack') isViable = (toID(set.species) === 'hawlucha');
 				if (id === 'smackdown') isViable = (template.types.indexOf('ground') > 0);
 				if (id === 'smartstrike') isViable = (template.types.indexOf('steel') > 0 && moves.indexOf('ironhead') < 0);
-				if (id === 'solarbeam') isViable = (toID(set.abilities) in {drought:1, chlorophyll:1});
+				if (id === 'solarbeam' || id === 'solarblade') isViable = (toID(set.ability) in {desolateland:1, drought:1, chlorophyll:1});
 				if (id === 'stompingtantrum') isViable = ((moves.indexOf('earthquake') < 0 && moves.indexOf('drillrun') < 0) || (toID(set.ability) === 'toughclaws' && moves.indexOf('drillrun') < 0 && moves.indexOf('earthquake') < 0));
 				if (id === 'storedpower') isViable = (toID(set.species) in {necrozma:1, espeon:1, sigilyph:1});
 				if (id === 'stunspore') isViable = (moves.indexOf('thunderwave') < 0);
+				if (id === 'teleport') isViable = (this.gen > 7);
```
Comment (Zarel) [inline js/search.js]: Drypass is extremely rare... how many people actually use Teleport competitively?

File: js/search.js [hunk 3]
```diff
@@ -1121,10 +1124,11 @@
 				if (id === 'skyattack') isViable = (toID(set.species) === 'hawlucha');
 				if (id === 'smackdown') isViable = (template.types.indexOf('ground') > 0);
 				if (id === 'smartstrike') isViable = (template.types.indexOf('steel') > 0 && moves.indexOf('ironhead') < 0);
-				if (id === 'solarbeam') isViable = (toID(set.abilities) in {drought:1, chlorophyll:1});
+				if (id === 'solarbeam' || id === 'solarblade') isViable = (toID(set.ability) in {desolateland:1, drought:1, chlorophyll:1});
```
Comment (Zarel) [inline js/search.js]:
```suggestion
				if (id === 'solarbeam' || id === 'solarblade') isViable = ['desolateland', 'drought', 'chlorophyll'].includes(toID(set.ability));
```

File: js/search.js (hunk 1 above)
Comment (Zarel) [reply js/search.js]: nvm, apparently it fails unless the user is Morpeko.

Weird restriction.

File: js/search.js (hunk 2 above)
Comment (KrisXV) [reply js/search.js]: people use it as a slow pivoting move to get a guaranteed switch into a pokemon that a teammate checks

Comment (Zarel) [reply js/search.js]: I guess I won't judge. PS's standards for usable moves are supposed to be really low.

File: js/search.js (hunk 1 above)
Comment (Zarel) [reply js/search.js]: I would still flag it in `moves.js`, for the record.

**Pull request merged**